src/core/qgsfield.h

```diff
@@ -66,13 +66,15 @@ class CORE_EXPORT QgsField
      * @param prec Field precision. Usually decimal places but may also be
      * used in conjunction with other fields types (eg. variable character fields)
      * @param comment Comment for the field
+     * @param subType If the field is a collection, its element's type
      */
     QgsField( const QString& name = QString(),
               QVariant::Type type = QVariant::Invalid,
               const QString& typeName = QString(),
               int len = 0,
               int prec = 0,
-              const QString& comment = QString() );
+              const QString& comment = QString(),
+              QVariant::Type subType = QVariant::Invalid );
```

Comment:
Just wondering - what about support for mixed type lists? I realise this may not be usable in Postgres, but other providers may support this and also I can see the value of having support for mixed type lists in the expression engine. Also support for lists of user types (eg QgsInterval, QgsGeometry) would be very useful.